module/VuFind/src/VuFind/Controller/CartController.php

```diff
@@ -48,23 +49,45 @@
  */
 class CartController extends AbstractBase
 {
+    use Feature\BulkActionControllerTrait;
+
     /**
      * Session container
      *
      * @var \Laminas\Session\Container
      */
     protected $session;
 
+    /**
+     * Configuration loader
+     *
+     * @var \VuFind\Config\PluginManager
+     */
+    protected $configLoader;
+
+    /**
+     * Export support class
+     *
+     * @var \VuFind\Export
+     */
+    protected $export;
```

Comment:
Would it make more sense to move this property definition into the trait that uses it, rather than duplicating it in multiple places?

Reply:
\VuFind\Export is actually also used in CartController.php. I removed the getExport method since we set it in the constructor now.

I checked how other traits handle dependencies and those that I checked state the dependency in the doc comment but don't define the properties themselves. And since they get set in the constructors of the classes I think it is better to define them in the classes as well. But this more based on a feeling than on knowledge of trait design concepts. If you prefer to define `$export` in the trait I will do so.

Reply:
It's fine to leave it as-is. This is the part of traits that I simply do not like -- I wish that traits could formally define requirements and/or have some kind of constructor equivalent to initialize them... but the reality is that traits are a very simplistic mechanism, and these types of things don't really have a place there. We just have to do our best to come up with useful conventions to avoid problems.

Perhaps there is an argument for moving the export property definition into the trait and also adding an `initializeExportTrait` method which could be called from the class constructors to provide the required value(s). But that feels like inventing a whole new mechanism that doesn't already have a precedent, and now is probably not the time to do it. :-)